File: client/src/types/logistics.ts
import { VehicleStatus } from "./common";

export interface Vehicle {
  id: number;
  vehicle_number: string;
  driver_name: string;
  status: VehicleStatus;
  orders_count: number;
  live_temp: number;
  fuel_level: number;
  latitude: number;
  longitude: number;
}

export interface AppSetting {
  setting_key: string;
  setting_value: string;
}

export interface AppSettingsUpdate {
  settings: AppSetting[];
}

File: server/app/schemas/admin/logistics.py
from pydantic import BaseModel, ConfigDict

class VehicleBase(BaseModel):
    vehicle_number: str
    driver_name: str
    latitude: float
    longitude: float
    status: str
    live_temp: float
    orders_count: int
    fuel_level: float

class Vehicle(VehicleBase):
    id: int
    model_config = ConfigDict(from_attributes=True)